isotptun.c

```diff
@@ -154,9 +177,9 @@ int main(int argc, char **argv)
 			else if (elements == 2)
 				opts.flags |= (CAN_ISOTP_EXTEND_ADDR | CAN_ISOTP_RX_EXT_ADDR);
 			else {
-				printf("incorrect extended addr values '%s'.\n", optarg);
+				fprintf(stderr, "incorrect extended addr values '%s'.\n", optarg);
 				print_usage(basename(argv[0]));
-				exit(0);
+				exit(EXIT_FAILURE);
 			}
```

Comment:
Since the use of it as a daemon requires changing some of the earlier success codes (0) to failures (EXIT_FAILURE) I thought they go logically into the same commit. Whoever is starting the daemon needs to know if it the daemon has started or if there was some error preventing it from starting. When it was only foreground, the exit code did not matter so much.